cmd/kubeadm/app/preflight/checks.go

```diff
@@ -1092,15 +1092,41 @@ func TryStartKubelet(ignorePreflightErrors sets.String) {
 	// If we notice that the kubelet service is inactive, try to start it
 	initSystem, err := initsystem.GetInitSystem()
 	if err != nil {
-		glog.Infoln("[preflight] no supported init system detected, won't ensure kubelet is running.")
-	} else if initSystem.ServiceExists("kubelet") {
-
-		glog.Infoln("[preflight] Activating the kubelet service")
-		// This runs "systemctl daemon-reload && systemctl restart kubelet"
-		if err := initSystem.ServiceRestart("kubelet"); err != nil {
-			glog.Warningf("[preflight] unable to start the kubelet service: [%v]\n", err)
-			glog.Warningf("[preflight] please ensure kubelet is running manually.")
-		}
+		glog.Infoln("[preflight] no supported init system detected, won't make sure the kubelet is running properly.")
+		return
+	}
+
+	if !initSystem.ServiceExists("kubelet") {
+		glog.Infoln("[preflight] couldn't detect a kubelet service, can't make sure the kubelet is running properly.")
+	}
+
+	glog.Infoln("[preflight] Activating the kubelet service")
+	// This runs "systemctl daemon-reload && systemctl restart kubelet"
+	if err := initSystem.ServiceRestart("kubelet"); err != nil {
+		glog.Warningf("[preflight] unable to start the kubelet service: [%v]\n", err)
+		glog.Warningf("[preflight] please ensure kubelet is reloaded and running manually.")
+	}
+}
+
+// TryStopKubelet attempts to bring down the kubelet service momentarily
+func TryStopKubelet(ignorePreflightErrors sets.String) {
+	if setHasItemOrAll(ignorePreflightErrors, "StopKubelet") {
+		return
+	}
+	// If we notice that the kubelet service is inactive, try to start it
+	initSystem, err := initsystem.GetInitSystem()
+	if err != nil {
+		glog.Infoln("[preflight] no supported init system detected, won't make sure the kubelet not running for a short period of time while setting up configuration for it.")
+		return
+	}
+
+	if !initSystem.ServiceExists("kubelet") {
+		glog.Infoln("[preflight] couldn't detect a kubelet service, can't make sure the kubelet not running for a short period of time while setting up configuration for it.")
+	}
+
+	// This runs "systemctl daemon-reload && systemctl stop kubelet"
+	if err := initSystem.ServiceRestart("kubelet"); err != nil {
```

Comment:
Um then we should call this f(n) restart.

Reply:
oops should be `.ServiceStop` here. fixed